torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import posixpath
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+
+from torch.utils.data.datapipes.utils.common import match_masks
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+try:
+    import datasets
+except ImportError:
+    datasets = None
+
+def _get_response_from_huggingface_hub(dataset, split, revision, streaming, data_files) -> Tuple[Any, StreamWrapper]:
+    dataset = datasets.load_dataset(dataset, split, revision, streaming, data_files)
+    return dataset[0], StreamWrapper(dataset)
```

Comment:
this is not a comment but more like a learning question for me:
What is StreamWrapper used for? dataset from `load_dataset` is already an iterable, could we return dataset or iter(dataset) directly?

Reply:
We seem to use it a bunch in the torchdata codebase to create iterable objects like so https://github.com/pytorch/data/blob/main/torchdata/datapipes/iter/load/online.py#L46

That said @NivekT had similar feedback to yours so maybe we can get rid of it

Reply:
We typically use `StreamWrapper` to wrap around streams that need to be closed during deletion.

https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes/utils/common.py#L172-L176

I am not familiar with HF dataset to be certain, my guess is that `StreamWrapper` isn't needed.